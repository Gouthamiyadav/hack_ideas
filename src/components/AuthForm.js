import React, { useState } from "react";
import { withStyles } from "@mui/styles";
import Button from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import { Paper, Typography } from "@mui/material";

const styles = {
  container: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "100vh",
    background: "#f0f0f0",
  },
  paperStyle: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    padding: "32px",
    background: "#fff",
    width: "400px",
    height: "auto",
    borderRadius: "8px",
    boxShadow: "0px 4px 10px rgba(0, 0, 0, 0.1)",
  },
  text: {
    fontFamily: "Inter",
    color: "#667085",
    fontSize: "24px",
    fontWeight: 600,
    marginBottom: "32px",
  },
};

const AuthForm = ({ classes, onLogin }) => {
  const [employeeId, setEmployeeId] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    onLogin(employeeId);
  };

  return (
    <div className={classes.container}>
      <Paper className={classes.paperStyle}>
        <Typography
          component="div"
          sx={{
            fontFamily: "Inter",
            color: "#00A0DFFF",
            fontSize: "30px",
            fontWeight: 600,
            marginBottom: "20px",
          }}
        >
          HACKATHON
        </Typography>
        <Typography
          component="div"
          sx={{
            fontFamily: "Inter",
            color: "#00000099",
            fontSize: "16px",
            fontWeight: 400,
            marginBottom: "20px",
          }}
        >
          Hack Ideas - Organize Your Internal Hackathons
        </Typography>
        <TextField
          sx={{ width: "100%", marginBottom: "16px" }}
          type="text"
          value={employeeId}
          onChange={(e) => setEmployeeId(e.target.value)}
          label="Enter Employee ID"
        />
        <Button
          sx={{ width: "100%", marginTop: "14px", marginBottom: "20px" }}
          variant="contained"
          color="primary"
          onClick={handleSubmit}
        >
          Login
        </Button>
      </Paper>
    </div>
  );
};

export default withStyles(styles, { withTheme: true })(AuthForm);
