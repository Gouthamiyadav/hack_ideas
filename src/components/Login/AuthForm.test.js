import React from "react";
import renderer from "react-test-renderer";
import AuthForm from "./AuthForm";
import { render } from "@testing-library/react";

jest.mock("@mui/styles", () => ({
  withStyles: (styles) => (Component) => (props) =>
    <Component {...props} classes={{ root: {} }} />,
  makeStyles: (styles) => (Component) => (props) =>
    <Component {...props} classes={{ root: {} }} />,
}));

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  useNavigate: () => mockNavigate,
}));

describe("AuthForm component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<AuthForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it("calls onLogin and navigates to /home when the form is submitted", async () => {
  const onLoginMock = jest.fn();
  const navigateMock = jest.fn();
  const { getByLabelText, getByText } = render(
    <AuthForm onLogin={onLoginMock} navigate={navigateMock} />
  );

  const employeeIdInput = getByLabelText("Enter Employee ID");
  fireEvent.change(employeeIdInput, { target: { value: "123456" } });

  const loginButton = getByText("Login");
  fireEvent.click(loginButton);

  await waitFor(() => {
    expect(onLoginMock).toHaveBeenCalledWith("123456");
    expect(navigateMock).toHaveBeenCalledWith("/home");
  });
});
