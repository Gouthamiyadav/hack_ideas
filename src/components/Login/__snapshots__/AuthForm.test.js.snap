// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm component should render correctly 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 css-cslod2-MuiTypography-root"
    >
      HACKATHON
    </div>
    <div
      className="MuiTypography-root MuiTypography-body1 css-1qosdk4-MuiTypography-root"
    >
      Hack Ideas - Organize Your Internal Hackathons
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root css-1ndnkrc-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Enter Employee ID
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Enter Employee ID
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-we1wxx-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Login
    </button>
  </div>
</div>
`;
